### DISCUSSION TITLE
Kafka mirror maker2 topic replication without source cluster name

### DISCUSSION THREAD

**Discussion by subash-subramaniyan** [strimzi/strimzi-kafka-operator#7489] [Q&A] [answered]
Im using kafka cluster KafkaMirrorMaker2 yaml file to syn the cluster topics to secondary cluster, But it copying the topic with source cluster name , I need to replicate the index as it in source cluster to secondary cluster please help me.

eg: mytoipc  ----> source.mytopic

apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaMirrorMaker2
metadata:
  name: my-mm2-cluster
spec:
  version: 3.2.0
  replicas: 1
  connectCluster: source
  clusters:
    - alias: source
      bootstrapServers: 'source-kafka-plain-bootstrap.test.in:9092'
    - alias: destination
      bootstrapServers: 'destination-kafka-plain-bootstrap.test.in:9092'
      config:
        config.storage.replication.factor: -1
        offset.storage.replication.factor: -1
        status.storage.replication.factor: -1
  mirrors:
    - sourceCluster: source
      targetCluster: destination
      sourceConnector:
        config:
          replication.factor: 1
          offset-syncs.topic.replication.factor: 1
          sync.topic.acls.enabled: 'false'
          replication.policy.class: org.strimzi.kafka.connect.mirror.IdentityReplicationPolicy
      heartbeatConnector:
        config:
          heartbeats.topic.replication.factor: 1
      checkpointConnector:
        config:
          checkpoints.topic.replication.factor: 1
      topicsPattern: .*
      groupsPattern: .*

**Comment by scholzj** [comment]
It is hard to say if your YAML is correct when it is not formatted properly. You can also compare it your self against the example (for your Strimzi version) -> there is IIRC an example for the replication with the same names.

**Reply by subash-subramaniyan** [reply]
Hi Scholzj , Im not able formate the yamls, Please find the attached file
[mm2.txt](https://github.com/strimzi/strimzi-kafka-operator/files/9800210/mm2.txt)

I have checked the most of doc not able to find the IIRC , Can you please share if you have any refernce

**Reply by scholzj** [reply]
The YAMl looks good in general. So I' not sure what the problem is. Depending on the version of Strimzi you use, you can check the example and apply the differences: https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/mirror-maker/kafka-mirror-maker-2-custom-replication-policy.yaml

**Reply by subash-subramaniyan** [reply]
Thanks for the information ,In 3..2.3 version , Kafka and zookeeper pods are not running , If we change version 3.2 its working

**Reply by scholzj** [reply]
I'm not sure how that is related ... and without the configs and logs, I will anyway not be able to help.

**Reply by subash-subramaniyan** [reply]
No issue, I will try to deploy with latest version and update the status , Thanks for the continues support

**Reply by subash-subramaniyan** [reply]
@scholzj  able to deploy with latest helm charts and topics syn properly , But active-active syn create multiple message both cluster, 
Please advise on this

![image](https://user-images.githubusercontent.com/93960578/196227353-cea33e6d-9f14-425f-b6a1-03feef4885ac.png)

**Comment by scholzj** [comment, marked as answer]
You cannot have both. You either need to have the topics renamed and then
you can use active-active mirroring. Or have the anmes kept and the. You
should mirror only in one direction.

**Reply by subash-subramaniyan** [reply]
Sure , Is it possible to delete attached topics and how to delete multiple topics in single shot
![image](https://user-images.githubusercontent.com/93960578/196229049-bffe74f2-2732-4807-ac48-bdc04ba349de.png)

**Comment by scholzj** [comment]
Sorry, I don't follow the question
